crates/web5/src/credentials/sign.rs

```diff
@@ -1,23 +1,26 @@
-use std::{sync::Arc, time::SystemTime};
-
+use super::{
+    jwt_payload_vc::JwtPayloadVerifiableCredential, verifiable_credential_1_1::VerifiableCredential,
+};
 use crate::{
-    crypto::dsa::Signer,
     dids::bearer_did::BearerDid,
     errors::{Result, Web5Error},
+    jose::{Jwt, JwtClaims},
+    json::{JsonValue, ToJsonValue},
 };
-use josekit::{jws::JwsHeader, jwt::JwtPayload};
-
-use super::{
-    josekit::JoseSigner, jwt_payload_vc::JwtPayloadVerifiableCredential,
-    verifiable_credential_1_1::VerifiableCredential,
-};
+use std::{collections::HashMap, time::SystemTime};
 
-pub fn sign_with_signer(
```

Comment:
did we remove the functionality to sign a credential with an arbitrary Signer?

Reply:
yeah, we weren't exposing it, and we are starting to bump up against poor design, because really we should probably add a `JoseSigner` which includes JOSE header values, otherwise we'd have to add an additional param here for the `Dsa` to be set as the `alg`